include/aws/common/byte_order.h

```diff
@@ -84,6 +84,44 @@ AWS_STATIC_IMPL uint32_t aws_ntoh32(uint32_t x) {
 #endif
 }
 
+/**
+ * Convert 24 bit integer from host to network byte order.
+ */
+AWS_STATIC_IMPL uint32_t aws_hton24(uint32_t x) {
+    AWS_PRECONDITION(x < (UINT32_MAX >> 8));
+    if (aws_is_big_endian()) {
+        return x;
+    } else {
+        uint32_t result = 0;
+        uint8_t *result_ptr = (uint8_t *)&result;
+
+        *result_ptr++ = (x >> 16) & 0xFF;
+        *result_ptr++ = (x >> 8) & 0xFF;
+        *result_ptr++ = (x >> 0) & 0xFF;
+
+        return result;
```

Comment:
is this the same as aws_ntoh32 followed by an 8 bit right shift?

Reply:
No you're right, it's exactly the same! Re-implementing...